### ISSUE TITLE
DeployVM UI Form: it passes default hypervisor as part of the call

### ISSUE THREAD

**Issue by borisstoyanov** [apache/cloudstack#6926]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.18
~~~

##### OS / ENVIRONMENT

Zone with multiple hypervisors 

##### SUMMARY

When trying to deploy a VM with hypervisor != default zone hypervisor, UI still passes the default one and fails the API call. 

##### STEPS TO REPRODUCE

~~~
In a KVM defualt zone, create a VM with xen template
~~~

##### EXPECTED RESULTS

~~~
API should not be passing the default hypervisor
~~~

##### ACTUAL RESULTS

~~~
It is, and therefore failing the call 
~~~

[Screenshot 2021-11-15 at 11.04.33[51].pdf](https://github.com/apache/cloudstack/files/10111250/Screenshot.2021-11-15.at.11.04.33.51.pdf)

```
Request Data
MIME Type: application/x-www-form-urlencoded;charset=utf-8
zoneid: 42bf38a6-f8a8-4e87-ad58-504b0f8dee47
clusterid: 8a9cc7c8-1a27-4a5e-9f98-39c142d9feb9
hostid: c4bdc0ad-9299-43a4-94ff-015ca4d8c756
boottype: BIOS
bootmode: LEGACY
templateid: d581c79e-6cc1-11ed-b2d6-1e0028000214
hypervisor: KVM
startvm: true
serviceofferingid: 952f1ffe-5bd2-4daa-b568-596369a640fa
affinitygroupids
iptonetworklist[0].networkid: f78ae6e1-dd47-42db-a350-4a7b5a1c1b85
keypairs
name: xen
displayname: xen
```

```
{
    "deployvirtualmachineresponse": {
        "uuidList": [],
        "errorcode": 431,
        "cserrorcode": 4350,
        "errortext": "Hypervisor passed to the deployVm call, is different from the hypervisor type of the template"
    }
}
```

**Comment by rohityadavcloud** [comment]
cc @rahulbcn27 may be a good minor issue to fix

**Comment by weizhouapache** [comment]
@rohityadavcloud @rahulbcn27 
this seems same as #6756 which is fixed by #6952

**Comment by DaanHoogland** [comment]
closing as per @weizhouapache 's comment. please reopen if this is still an issue @rahulbcn27 @borisstoyanov
